File: pset6/vigenere/vigenere.py
import sys
from cs50 import get_string

if len(sys.argv) != 2:
    print("ERROR")
    exit(1)

k = sys.argv[1]
m = len(k)
for c in k:
    if not(c.isalpha()):
        print("ERROR")
        exit(1)

p = get_string("plaintext:")
print("ciphertext:", end="")
j = 0

for c in p:
    if c.isalpha():
        if c.isupper():
            if k[j % m].isupper():
                c = chr(65 + (ord(c) - 65 + ord(k[j % m]) - 65) % 26)
                print(c, end="")
            if k[j % m].islower():
                c = chr(65 + (ord(c) - 65 + ord(k[j % m]) - 97) % 26)
                print(c, end="")
        elif c.islower():
            if k[j % m].isupper():
                c = chr(97 + (ord(c) - 97 + ord(k[j % m]) - 65) % 26)
                print(c, end="")
            if k[j % m].islower():
                c = chr(97 + (ord(c) - 97 + ord(k[j % m]) - 97) % 26)
                print(c, end="")
        j += 1
    else:
        print(c, end="")
print()

File: pset2/vigenere/vigenere.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    //only accept two command lines
    if (argc != 2)
    {
        printf("ERROR");
        return 1;
    }

    //store key
    string k = argv[1];

    //make sure key is alpha
    for (int i = 0, n = strlen(k); i < n; i++)
    {
        if (!isalpha(k[i]))
        {
            printf("ERROR");
            return 1;
        }
    }

    int m = strlen(k);
    string p = get_string("plaintext:");
    printf("ciphertext:");
    int j = 0;

    for (int i = 0, n = strlen(p); i < n; i++)
    {
        //use key when plaintext is alpha
        if (isalpha(p[i]))
        {
            //if plaintext is upper
            if (isupper(p[i]))
            {
                //if key is upper
                if (isupper(k[j % m]))
                {
                    p[i] = 65 + (p[i] - 65 + k[j % m] - 65) % 26;
                    printf("%c", p[i]);
                }
                //if key is lower
                if (islower(k[j % m]))
                {
                    p[i] = 65 + (p[i] - 65 + k[j % m] - 97) % 26;
                    printf("%c", p[i]);
                }
            }

            //if plaintext is lower
            else if (islower(p[i]))
            {
                //if key is upper
                if (isupper(k[j % m]))
                {
                    p[i] = 97 + (p[i] - 97 + k[j % m] - 65) % 26;
                    printf("%c", p[i]);
                }
                //if key is lower
                if (islower(k[j % m]))
                {
                    p[i] = 97 + (p[i] - 97 + k[j % m] - 97) % 26;
                    printf("%c", p[i]);
                }
            }
            j++;
        }
        //when plaintext is not alpha
        else
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
}